cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaBrokerConfigurationBuilder.java

```diff
@@ -157,23 +198,29 @@ public KafkaBrokerConfigurationBuilder withZookeeper(String clusterName)  {
      * @param kafkaListeners                The listeners configuration from the Kafka CR
      * @param controlPlaneListenerActive    Activates the control plane listener (the listener is always configured,
      *                                      but this flag tells Kafka to use it for control plane communication)
+     * @param escapeTheZoo                  Activates the Zookeeper-less mode
      *
      * @return  Returns the builder instance
      */
-    public KafkaBrokerConfigurationBuilder withListeners(String clusterName, String namespace, List<GenericKafkaListener> kafkaListeners, boolean controlPlaneListenerActive)  {
+    public KafkaBrokerConfigurationBuilder withListeners(String clusterName, String namespace, List<GenericKafkaListener> kafkaListeners, boolean controlPlaneListenerActive, boolean escapeTheZoo)  {
         List<String> listeners = new ArrayList<>();
         List<String> advertisedListeners = new ArrayList<>();
         List<String> securityProtocol = new ArrayList<>();
 
         // Control Plane listener
         listeners.add("CONTROLPLANE-9090://0.0.0.0:9090");
-        advertisedListeners.add(String.format("CONTROLPLANE-9090://%s:9090",
-                // Pod name constructed to be templatable for each individual ordinal
-                DnsNameGenerator.podDnsNameWithoutClusterDomain(namespace, KafkaResources.brokersServiceName(clusterName),
-                        KafkaResources.kafkaStatefulSetName(clusterName) + "-${STRIMZI_BROKER_ID}")
-        ));
-        securityProtocol.add("CONTROLPLANE-9090:SSL");
-        configureControlPlaneListener();
+
+        if (!escapeTheZoo) {
+            advertisedListeners.add(String.format("CONTROLPLANE-9090://%s:9090",
+                    // Pod name constructed to be templatable for each individual ordinal
+                    DnsNameGenerator.podDnsNameWithoutClusterDomain(namespace, KafkaResources.brokersServiceName(clusterName),
+                            KafkaResources.kafkaStatefulSetName(clusterName) + "-${STRIMZI_BROKER_ID}")
+            ));
+            securityProtocol.add("CONTROLPLANE-9090:SSL");
+            configureControlPlaneListener();
+        } else {
+            securityProtocol.add("CONTROLPLANE-9090:PLAINTEXT");
```

Comment:
I guess for each of these places where we're disabling functionality we should raise an issue so we can tackle it bit by bit and not lose track of temporarily missing features in kraft mode. Adding the issue number in the code is probably also worthwhile in this case.

Reply:
I'm not sure this is something what should evolve into production implementation - so I'm not sure it makes sense to _tackle it bit by bit_. I guess having some comments pointing that out might be good.